src/server/ua_services_discovery.c

```diff
@@ -502,6 +502,11 @@ process_RegisterServer(UA_Server *server, UA_Session *session,
     if(!dm)
         return;
 
+    if(dm->serverConfig->applicationDescription.applicationType != UA_APPLICATIONTYPE_DISCOVERYSERVER) {
```

Comment:
Add this to the `UA_ServerConfig` documentation for the `applicationDescription`.
The `applicationDescription` is not only descriptive, but changes the actual server functionality.